mypy/semanal.py

```diff
@@ -1428,7 +1438,12 @@ def add_unknown_symbol(self, name: str, context: Context, is_import: bool = Fals
         else:
             var._fullname = self.qualified_name(name)
         var.is_ready = True
-        var.type = AnyType()
+        any_type = AnyType()
+        if self.options.silent_mode():
+            # if silent mode is not enabled, no need to report implicit conversion to Any,
+            # let mypy report an import error.
```

Comment:
Yeah, this makes sense. It's not obvious that the new flag should be used with the silent imports.